lib/commons/color/get-text-shadow-colors.js

```diff
@@ -1,18 +1,27 @@
 import Color from './color';
 import assert from '../../core/utils/assert';
 
+/** Magic numbers **/
+// Alpha value to use when text shadows are offset between .5px and 1.5px
+const SHADOW_STROKE_ALPHA = 0.54;
+// Shadows offset by less than this are not visible enough no matter how much you stack them
+const VISIBLE_SHADOW_MIN_PX = 0.5;
+// Shadows offset by more than this have full opacity
+const OPAQUE_STROKE_OFFSET_MIN_PX = 1.5;
+
 /**
  * Get text-shadow colors that can impact the color contrast of the text
```

Comment:
It would be nice for the doc comment to summarize the total behavior

```suggestion
 * Get text-shadow colors that can impact the color contrast of the text, including from:
 * - Shadows which are individually thick enough to distinguish as characters
 * - Groups of thin shadows that collectively act as a pseudo-text-stroke (see #4064)
```